restricted_site_access.php

```diff
@@ -209,7 +221,9 @@ public static function restrict_access( $wp ) {
 			$blog_public = get_site_option( 'blog_public', 2 );
 		}
 
-		$is_restricted = !( is_admin() || is_user_logged_in() || 2 != $blog_public || ( defined( 'WP_INSTALLING' ) && isset( $_GET['key'] ) ) );
+		$is_restricted = self::is_restricted( $blog_public );
```

Comment:
@ptasker Instead of passing `$blog_public` here, I'd move the code that generates that variable into the `is_restricted` method, as that variable isn't used anywhere else. This also means you'd need to move the code that gets the `$mode` variable, but that variable also isn't used anywhere else, so that should be fine.